### PR TITLE
[KYUUBI #1417] Parse instance information in a compatible way

### PR DESCRIPTION
**Pull request by wForget**
### _Why are the changes needed?_

Fix bug #1417.

After fix, run the `bin/kyuubi-ctl list server` command successfully.

image

### _How was this patch tested?_
- [X] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [X] Add screenshots for manual tests if appropriate

- [X] [Run test](https://kyuubi.readthedocs.io/en/latest/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

**Review by yaooqinn — APPROVED** [review]

File: kyuubi-ha/src/main/scala/org/apache/kyuubi/ha/client/ServiceDiscovery.scala [hunk 1]
```diff
@@ -205,6 +204,23 @@ object ServiceDiscovery extends Logging {
...
+    if (maybeInfos.size > 0) {
+      (maybeInfos.get("hive.server2.thrift.bind.host").get,
+        maybeInfos.get("hive.server2.thrift.port").get.toInt)
+    } else {
+      val strings = instance.split(":")
+      (strings(0), strings(1).toInt)
+    }
+  }
+
+
```
Comment (ulysses-you) [inline kyuubi-ha/src/main/scala/org/apache/kyuubi/ha/client/ServiceDiscovery.scala:223]: nit: more one line

**Review by ulysses-you — APPROVED** [review]

**Comment by pan3793** [comment]
Also cc @cxzl25

**Comment by pan3793** [comment]
Thanks, merging to master

File: kyuubi-ha/src/main/scala/org/apache/kyuubi/ha/client/ServiceDiscovery.scala (hunk 1 above)
Comment (cxzl25) [reply kyuubi-ha/src/main/scala/org/apache/kyuubi/ha/client/ServiceDiscovery.scala:223]: LGTM

Hive uses regular expressions to be compatible.

https://issues.apache.org/jira/browse/HIVE-13326

org.apache.hive.jdbc.ZooKeeperHiveClientHelper#configureConnParams
```java
  private static final Pattern kvPattern = Pattern.compile("([^=;]*)=([^;]*)[;]?");


      Matcher matcher = kvPattern.matcher(dataStr);
      // If dataStr is not null and dataStr is not a KV pattern,
      // it must be the server uri added by an older version HS2
      if ((dataStr != null) && (!matcher.find())) {
        String[] split = dataStr.split(":");
        if (split.length != 2) {
          throw new ZooKeeperHiveClientException("Unable to read HiveServer2 uri from ZooKeeper: "
              + dataStr);
        }
        connParams.setHost(split[0]);
        connParams.setPort(Integer.parseInt(split[1]));
```

**Pull request closed without merging**